modules/api/php/endpoints/candidate/visit/dicoms.class.inc

```diff
@@ -138,20 +156,167 @@ class Dicoms extends Endpoint implements \LORIS\Middleware\ETagCalculator
         case "v0.0.3":
             $view = (new \LORIS\api\Views\Visit\Dicoms(
                 $this->_visit,
-                ...$dicomtars
+                $dicomtars
             ))->toArray();
             break;
         default:
-            $view = (new \LORIS\api\Views\Visit\Dicoms_0_0_4_Dev(
+            $view = (new \LORIS\api\Views\Visit\Dicoms(
                 $this->_visit,
-                ...$dicomtars
+                $dicomtars
             ))->toArray();
         }
         $this->_cache = new \LORIS\Http\Response\JsonResponse($view);
 
         return $this->_cache;
     }
 
+    /**
+     * Upload the DICOM study file and add it to the timepoint's MRIUploads
+     * A successful upload should return a 303 response toward this DICOM
+     * processes.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request): ResponseInterface
+    {
+        // *************************
+        // *** Permission checks ***
+        // *************************
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('imaging_uploader')) {
+            return new \LORIS\Http\Response\JSON\Forbidden();
+        }
+
+        // *************************
+        // *** Input Validation ****
+        // *************************
+        $files = $request->getUploadedFiles();
+        if (count($files) !== 1) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'There should be one file uploaded per POST request.'
+            );
+        }
+
+        if (!isset($files['MriFile'])) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'The input key for the uploaded file should be MriFile.'
```

Comment:
Note: most use cases we have are MRI files but technically, users could upload PET DICOM files as well, so I would tend to keep the 'MRI' string out of the docs and JSON responses and instead use the more generic term "imaging file".

Reply:
Hhhhm, I changed the naming of the two form-data fields to just `Json` and `File`. The documentation then goes into more details about what are the roles and structures of the expected data.